modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/ExtensionComponent.java

```diff
@@ -158,7 +159,7 @@
 
   protected MetadataCacheIdGenerator<ComponentAst> cacheIdGenerator;
 
-  private Function<CoreEvent, Optional<ConfigurationInstance>> configurationResolver;
+  private BiFunction<CoreEvent, Boolean, Optional<ConfigurationInstance>> configurationResolver;
```

Comment:
I don't think this added parameter is a good idea. Why not just injecting the MuleContext and using the isStopping() method?